community/detectors/selenium_grid_rce_via_exposed_server/src/main/java/com/google/tsunami/plugins/detectors/rce/selenium/RCEViaExposedSeleniumGridDetector.java

```diff
@@ -0,0 +1,523 @@
+
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+
+  private final Clock utcClock;
+  private final HttpClient httpClient;
+  private final PayloadGenerator payloadGenerator;
+  private final String payloadFormatString;
+  private final String seleniumUrlPayload;
+  private final String seleniumSessionSettings;
+  private static final String SELENIUM_GRID_SERVICE_PATH = "wd/hub";
+  private final String RCE_TEST_FILE_PATH = "/tmp/tsunami-selenium-rce";
+
+  @VisibleForTesting
+  static final String RCE_TEST_STRING =
+      "tsunami-selenium-rce-" + Long.toHexString(Double.doubleToLongBits(Math.random()));
+
+  // Selenium Grid ready state wait timeout. It's set to 310s (~5min) here.
+  // Default Selenium Grid in uses 300s timeouts so it should be more than this.
+  private final int POLLING_RATE = 10000; // 10s
+  private final int POLLING_ATTEMPTS = 31;
+
+  // Tsunami scanner relies heavily on Guice framework. So all the utility dependencies of your
+  // plugin must be injected through the constructor of the detector. Notably, the Payload
+  // generator is injected this way.
+  @Inject
+  RCEViaExposedSeleniumGridDetector(
+      @UtcClock Clock utcClock, HttpClient httpClient, PayloadGenerator payloadGenerator)
+      throws IOException {
+    this.utcClock = checkNotNull(utcClock);
+    // TODO: setReadTimeout() method is missing in Tsunami HttpClient.java.
+    // It is needed to avoid false negatives. See TODO(b/145315535) in tsunami-scanner
+    // Enable the line below once this bug has been fixed.
+    // this.httpClient = httpClient.modify().setReadTimeout(Duration.ofSeconds(15)).build();
+    this.httpClient = httpClient.modify().setConnectTimeout(Duration.ofSeconds(10)).build();
+    this.payloadGenerator = checkNotNull(payloadGenerator);
+
+    this.payloadFormatString =
+        String.format(
+            Resources.toString(
+                Resources.getResource(this.getClass(), "payloadFormatString.json"), UTF_8),
+            "%s"); // Placeholder for the command payload
+
+    this.seleniumSessionSettings =
+        Resources.toString(
+            Resources.getResource(this.getClass(), "payloadSessionSettings.json"), UTF_8);
+
+    this.seleniumUrlPayload =
+        String.format(
+            Resources.toString(
+                Resources.getResource(this.getClass(), "payloadSeleniumUrl.json"), UTF_8),
+            "%s"); // Placeholder for URL
+  }
+
+  // This is the main entry point of VulnDetector.
+  @Override
+  public DetectionReportList detect(
+      TargetInfo targetInfo, ImmutableList<NetworkService> matchedServices) {
+    logger.atInfo().log("RCEViaExposedSeleniumGridDetector starts detecting.");
+
+    return DetectionReportList.newBuilder()
+        .addAllDetectionReports(
+            matchedServices.stream()
+                // TODO: isWebService filter doesn't run this detector.
+                // .filter(NetworkServiceUtils::isWebService)
```

Comment:
The `example_vuln_detector` in the repository doesn't have the filter on `isWebService`.
I think these line can be deleted